lib/git_worktree.rb

```diff
@@ -336,7 +336,8 @@ def fetch_and_merge
 
   def fetch
     with_remote_options do |remote_options|
-      @repo.fetch(@remote_name, **remote_options)
+      fetch_options = remote_options.merge(prune: true)
```

Comment:
@jaisejose1123 what do you think about exposing this as an option to fetch, fetch_and_pull, and pull?  That should give the caller the option to not prune if they don't want to.

e.g.:
```
def fetch(prune: true)
```

Reply:
Looks good @jaisejose1123 we also need to add it to `fetch_and_merge` https://github.com/ManageIQ/manageiq/pull/23115/files#diff-d54d482e470404136b4b893c60065b5d743c912652d58d69820d86604e8a8983R331 and `pull` in order for a user to access the option (fetch is a private method)

Reply:
@agrare  updated the PR by adding the `prune` option in `fetch_and_merge` and `pull`. Please review.